modules/pam_filter/pam_filter.c

```diff
@@ -354,7 +354,7 @@ set_filter (pam_handle_t *pamh, int flags UNUSED, int ctrl,
 	    int t = open("/dev/tty", O_RDWR|O_NOCTTY);
 #else
 	    int t = open("/dev/tty",O_RDWR);
-	    if (t > 0) {
+	    if (t >= 0) {
 		(void) ioctl(t, TIOCNOTTY, NULL);
 		close(t);
 	    }
```

Comment:
This was not "just" an fd leak, but also a bug in the attempt to close the controlling tty.
I'd like to have it reflected in the commit message, too.

Reply:
Do you mean to add something like:
```
Fix bug when closing tty and descriptor leak

Failing to check the descriptor value meant that there was a bug in the attempt to close the controlling tty. Moreover, this would lead to a file descriptor leak.
```
Then, I would add the covscan report. What do you think?

Reply:
I've changed the commit message, can you take a look at it?